### ISSUE TITLE
ios detox execution hangs with 'failed to add line to log:'

### ISSUE THREAD

**Issue by vincenzovitale** [wix/Detox#1142]
#### Description
Detox ios execution hangs with:

detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:

being printed forever.

#### Steps to Reproduce
We were not able to reproduce it, it's not always happening and the app in the simulator runs fine, including all the detox step. But the detox execution hangs.

#### Detox, Node, Device, Xcode and macOS Versions

* Detox: 10.0.5
* React Native: 0.57.2
* Node: 8.9.4
* Device: iPhone X simulator
* Xcode: 10.1
* macOS: 10.14.2

#### Device and verbose Detox logs
[17:55:10][detox_ios]   console.log node_modules/jasmine-reporters/src/teamcity_reporter.js:37
[17:55:10][detox_ios]     
[17:55:10][detox_ios] 
[17:55:10][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:10][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:11][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:11][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:11][detox_ios] Jest did not exit one second after the test run has completed.
[17:55:11][detox_ios] 
[17:55:11][detox_ios] This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
[17:55:11][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:11][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:12][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:12][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:12][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:12][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:13][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:13][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:13][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:13][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:14][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:14][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:14][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:14][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:15][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:15][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:15][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:15][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:16][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:16][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:16][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:16][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:17][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:17][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:17][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:17][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:18][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:18][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:19][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:19][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"
[17:55:19][detox_ios] detox[7329] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
[17:55:19][detox_ios] ⚠️-[LocationManager locationManager:didFailWithError:] Error Domain=kCLErrorDomain Code=0 "(null)"

**Comment by LeoNatan** [comment]
@noomorph Please take a look.

**Comment by silyevsk** [comment]
I just got something similar (or maybe the same) - many lines like this:
```
detox[2093] WARN:  [SimulatorLogRecording.js/LOG_WRITE_ERROR] failed to add line to log:
  uuid: '14517e1a-3ff0-af98-408e-2bd6953c36a2'
```
The detox didn't stop and eventually printed that the tests passed, but then it got stuck (forever).

**Comment by LeoNatan** [comment]
@noomorph Please take a look.

**Comment by noomorph** [comment]
Fixed in https://github.com/wix/Detox/commit/65e1975c5164ee82658255f3ec9e44fdce412896 . Available since `detox@12.1.4`.
